src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,801 @@
+        {
+            if (socksCommandRequest.version() != 5 && socksCommandRequest.reserved() != 0)
+            {
+                //TODO
+            }
+
+            switch (socksCommandRequest.command().get())
+            {
+                case CONNECT:
+                    onSocksConnect(socksCommandRequest);
+                    break;
+                case BIND:
+                    //TODO
+                    break;
+                case UDP_ASSOCIATE:
+                    //TODO
+                    break;
+            }
+        }
+
+        private void onSocksConnect(
+            SocksCommandRequestFW socksCommandRequest)
+        {
+            final MessagePredicate filter = (t, b, o, l) ->
+            {
+                if (t > 0 && t <= ResetFW.TYPE_ID && o < l)
+                {
+                    return true;
+                }
+                return false;
+            };
+
+            final RouteFW route = router.resolve(routeId, this.authorization, filter, wrapRoute);
+            if(route != null)
+            {
+                final long newRouteId = route.correlationId();
+                final long newInitialId = supplyInitialId.applyAsLong(newRouteId);
+                final long newReplyId = supplyReplyId.applyAsLong(newInitialId);
+
+                final MessageConsumer newTarget = router.supplyReceiver(newInitialId);
+                final SocksServerStream socksServerStream = new SocksServerStream(this, newTarget,
+                    newRouteId, newInitialId, newReplyId);
+
+                SocksAddressFW socksAddress = socksCommandRequest.address();
+                String address = formatSocksAddress(socksAddress);
+
+                socksServerStream.doApplicationBegin(newTarget,
+                                                     newInitialId,
+                                                     decodeTraceId,
+                                                     address,
+                                                     socksCommandRequest.port());
+
+                correlations.put(newReplyId, socksServerStream);
+            }
+        }
+
+        private String formatSocksAddress(
+            SocksAddressFW socksAddress)
+        {
+            String address = "";
+            switch (socksAddress.kind())
+            {
+                case SocksAddressFW.KIND_DOMAIN_NAME:
+                    address = socksAddress.domainName().asString();
+                    break;
+                case SocksAddressFW.KIND_IPV4_ADDRESS:
+                    OctetsFW ipRO = socksAddress.ipv4Address();
+                    ipRO.buffer().getBytes(ipRO.offset(), writeBuffer, 0, ipRO.sizeof());
+                    address = ((long) writeBuffer.getByte(0) & 0xffL) + "." +
+                              ((long) writeBuffer.getByte(1) & 0xffL) + "." +
+                              ((long) writeBuffer.getByte(2) & 0xffL) + "." +
+                              ((long) writeBuffer.getByte(3) & 0xffL);
```

Comment:
Let's revisit this one to get rid of the string concatenation which produces garbage.
Instead of `String` we can perhaps use `StringFW`.